### PR TITLE
Update scalafmt to 3.0.6

### PR DESCRIPTION
**Pull request by armanbilge**
Scala 3 sources needed a dialect override because they are using Scala 3 only syntax that is not compatible with the `Scala213Source3` dialect.

### CODE REVIEW

File: benchmark/src/main/scala/spire/benchmark/ZigguratBenchmarks.scala [hunk 1]
```diff
@@ -1,15 +1,8 @@
...
- * *  /_____/ /_/      /_/   /_/  |_|  /____/     All rights reserved.    **
- * *                                                                      **
- * *      Redistribution and use permitted under the MIT license.         **
- * *                                                                      **
+ * **********************************************************************\ * Project ** * ______ ______ __ ______ ____
+ * ** * / ____/ / __ / / / / __ / / __/ (c) 2011-2014 ** * / /__ / /_/ / / / / /_/ / / /_ ** * /___ / / ____/ / / / __ /
+ * / __/ Erik Osheim, Tom Switzer ** * ____/ / / / / / / / | | / /__ ** * /_____/ /_/ /_/ /_/ |_| /____/ All rights
+ * reserved. ** * ** * Redistribution and use permitted under the MIT license. ** * **
  * \***********************************************************************
  */
```
Comment (armanbilge) [inline benchmark/src/main/scala/spire/benchmark/ZigguratBenchmarks.scala:7]: Oh no 🤦

Comment (armanbilge) [reply benchmark/src/main/scala/spire/benchmark/ZigguratBenchmarks.scala:7]: follow up in #1086

Comment (cquiroz) [reply benchmark/src/main/scala/spire/benchmark/ZigguratBenchmarks.scala:7]: I think you can indicate a block doesn't need formatting
https://scalameta.org/scalafmt/docs/configuration.html#-format-off

Comment (armanbilge) [reply benchmark/src/main/scala/spire/benchmark/ZigguratBenchmarks.scala:7]: Yes, but I've never seen anybody use that for their headers. Maybe this situation is different, but I'd prefer a better fix if possible.

File: core/src/main/scala-3/spire/syntax/Syntax.scala [hunk 2]
```diff
@@ -269,50 +271,50 @@ trait LiteralsSyntax {
       sb.append(lits.next())
       while (vars.hasNext)
         sb.append(vars.next())
-        sb.append(lits.next())
+      sb.append(lits.next())
```
Comment (armanbilge) [inline core/src/main/scala-3/spire/syntax/Syntax.scala:274]: Oh crap.

File: core/src/main/scala-3/spire/syntax/macros/cforMacros.scala [hunk 3]
```diff
@@ -94,54 +96,54 @@ def fastForRangeMacro(r: Expr[Range], body: Expr[Int => Unit])(using quotes: Quo
...
     }
+  }
 
   def strideUpTo(fromExpr: Expr[Int], untilExpr: Expr[Int], stride: Expr[Int]): Expr[Unit] = '{
     var index = $fromExpr
-    val end   = $untilExpr
+    val end = $untilExpr
     while (index <= end)
-      ${ Expr.betaReduce(body) }(index)
-      index += $stride
```
Comment (armanbilge) [inline core/src/main/scala-3/spire/syntax/macros/cforMacros.scala:110]: hmm??

Comment (cquiroz) [reply core/src/main/scala-3/spire/syntax/macros/cforMacros.scala:110]: I would expect `scalafmt` not to change the semantics of the code

Comment (armanbilge) [reply core/src/main/scala-3/spire/syntax/macros/cforMacros.scala:110]: Yeah seems like a bug. 😠

File: core/src/main/scala-3/spire/syntax/macros/cforMacros.scala [hunk 4]
```diff
@@ -3,24 +3,26 @@ package spire.syntax.macros
...
 
-import spire.syntax.fastFor.{RangeLike, RangeElem}
+import spire.syntax.fastFor.{RangeElem, RangeLike}
 
 inline def fastForInline[R](init: R, test: R => Boolean, next: R => R, body: R => Unit): Unit =
   var index = init
   while (test(index))
     body(index)
-    index = next(index)
+  index = next(index)
```
Comment (cquiroz) [inline core/src/main/scala-3/spire/syntax/macros/cforMacros.scala]: Hmm, this changes the code

File: core/src/main/scala-3/spire/syntax/Syntax.scala [hunk 5]
```diff
@@ -71,7 +71,9 @@ trait AdditiveSemigroupSyntax:
     new LiteralDoubleAdditiveSemigroupOps(lhs)
 
 trait AdditiveMonoidSyntax extends AdditiveSemigroupSyntax:
-  implicit def additiveMonoidOps[A](a: A)(implicit ev: AdditiveMonoid[A]): AdditiveMonoidOps[A] = new AdditiveMonoidOps(a)
+  implicit def additiveMonoidOps[A](a: A)(implicit ev: AdditiveMonoid[A]): AdditiveMonoidOps[A] = new AdditiveMonoidOps(
```
Comment (cquiroz) [inline core/src/main/scala-3/spire/syntax/Syntax.scala:74]: Maybe a line break?

**Pull request merged**